### ISSUE TITLE
训不出来

### ISSUE THREAD

**Issue by Play2Boy** [michuanhaohao/keras_reid#2]
表示训不出来

**Comment by mengxingzhi** [comment]
同训不出来，希望能给个解释

**Comment by michuanhaohao** [comment]
@Play2Boy @mengxingzhi 

softmax loss和triplet loss应该是很稳定的，msml受初始化影响可能训不出来，可以看下issue 1，给出了一个很好解决方案，可以参考一下

**Comment by kardoszc** [comment]
@michuanhaohao 感觉是数据里面有噪音就会那样，比如完全一样的图，开方做backward的时候会变0，然后就NaN了
@Play2Boy @mengxingzhi
softmax loss和triplet loss确实不会有这种问题，不收敛的话可以看看做数据是不是有点问题

**Comment by michuanhaohao** [comment]
@kardoszc 是的，我实验发现的cuhk03数据集有NAN这个现象，因为cuhk03每个人只有两张图，但是MSML一般要求每个人4张图，所以会出现重复的图出现。我在market抽图片可以保证选择不同的图，可以end2end很稳定地训练
